src/issues/github.rs

```diff
@@ -1,44 +1,79 @@
 
-pub(crate) async fn list_issues(
+const ISSUES_QUERY: &str = r##"
+query($repo: String!, $owner: String!, $labels: [String!]) { 
+  repository(name:$repo, owner:$owner) { 
+    issues(states:OPEN, first: 30, labels: $labels) {
+      nodes {
+        number,
+        title
+      }
+    }
+  }
+}
+"##;
+
+#[derive(serde::Deserialize)]
+struct ResponseIssue {
+    number: usize,
+    title: String,
+}
+
+pub(crate) fn list_issues(
     github_config: Option<config::GitHub>,
     github_state: state::GitHub,
-    labels: Option<Vec<String>>,
+    labels: &Option<Vec<String>>,
 ) -> Result<(Option<config::GitHub>, state::GitHub, Vec<Issue>)> {
     match github_config {
         None => Err(eyre!("GitHub is not configured")),
         Some(github_config) => {
-            let octocrab = match github_state {
-                state::GitHub::Initialized { octocrab } => octocrab,
-                state::GitHub::New => get_or_prompt_for_octocrab()?,
-            };
-            let request = octocrab.issues(&github_config.owner, &github_config.repo);
-            let oc_issues = if let Some(labels) = labels {
-                request.list().labels(labels.as_slice()).send().await?
-            } else {
-                request.list().send().await?
+            let token = match github_state {
+                state::GitHub::Initialized { token } => token,
+                state::GitHub::New => get_or_prompt_for_github_token()?,
             };
-            let issues = oc_issues
-                .items
+            let response = ureq::post("https://api.github.com/graphql")
+                .set("Authorization", &format!("bearer {}", &token))
+                .send_json(ureq::json!({
+                    "query": ISSUES_QUERY,
+                    "variables": {
+                        "repo": &github_config.repo,
+                        "owner": &github_config.owner,
+                        "labels": labels
+                    }
+                }))?;
+
+            let gh_issues = decode_github_response(response)?;
+
+            let issues = gh_issues
                 .into_iter()
-                .filter_map(|oc_issue| match oc_issue.pull_request {
-                    Some(_) => None,
-                    None => Some(Issue {
-                        key: oc_issue.number.to_string(),
-                        summary: oc_issue.title,
-                    }),
+                .map(|gh_issue| Issue {
+                    key: gh_issue.number.to_string(),
+                    summary: gh_issue.title,
                 })
                 .collect();
 
             Ok((
                 Some(github_config),
-                state::GitHub::Initialized { octocrab },
+                state::GitHub::Initialized { token },
                 issues,
             ))
         }
     }
 }
+
+fn decode_github_response(response: ureq::Response) -> Result<Vec<ResponseIssue>> {
+    let json_value = response.into_json::<serde_json::Value>()?;
```

Comment:
May consider using [`ureq::serde_to_json`](https://docs.rs/ureq/2.0.2/ureq/fn.serde_to_value.html) instead, which removes the need for the turbofish syntax.

Reply:
`Response` would have to impl `Serialize` for that to work, which it does not. However I did switch to type inference from turbofish since an earlier cleanup made that possible, if you think that's cleaner.

Reply:
I do think that's cleaner.